### ISSUE TITLE
PSF data grid does not seem to be centered correctly

### ISSUE THREAD

**Issue by andibarg** [MREYE-LUMC/ZOSPy#54]
### Problem

While using the Huygens PSF analysis for a rotationally symmetric system, I noticed that the maximum value of the PSF is not at (0,0). This is in contrast to the output directly taken from OpticStudio. I provided a simplified example below to reproduce this. And in case you are wondering, it seems to have nothing to do with the "use centroid" option.

As far as I can see, the problem originates from the `unpack_datagrid` function in `zputils.py`.  In row 104, we have:
```python
df = pd.DataFrame(data=values, index=rows[::-1], columns=columns)
``` 
I inverted the array `rows`, which gave me the correct results:
```python
df = pd.DataFrame(data=values, index=rows, columns=columns)
``` 
What is the reason for inverting `rows` in the first place? Do you see any issues with my solution? Is `unpack_datagrid` used by any other analysis that should be checked?

### What version of ZOSPy are you running?

1.1.1

### What version of OpticStudio are you running?

23.2.1

### Which operating system do you use?

Windows 10

### In which environment do you use ZOSPy?

- [X] Plain Pythons scripts (not in Spyder)
- [ ] Jupyter notebooks
- [ ] Spyder

### In which connection modes does the problem occur?

- [ ] Extension mode
- [X] Standalone mode

### Example code

```Python
import os
import numpy as np
import zospy as zp
import matplotlib.pyplot as plt

# Load Zemax file
zos = zp.ZOS()
oss = zos.create_new_application(True)
oss.load(os.path.join(zos.Application.SamplesDir,
                      'Sequential',
                      'Objectives',
                      'Double Gauss 28 degree field.zmx')

# Get Huygens PSF
psfdata = zp.analyses.psf.huygens_psf(oss).Data

# Get center and max value
print(psfdata[0][0])
print(np.max(psfdata))

# Plot
plt.pcolormesh(psfdata.keys(),psfdata[0].keys(),psfdata)
plt.show()
```

**Comment by crnh** [comment]
@LucVV do you have time to look into this and check if this problem also occurs in older versions of OpticStudio?

**Comment by crnh** [comment]
Hi @andibarg,

Thanks for reporting this problem! I investigated this issue and I think you are right. This issue probably arose because the PSF output by OpticStudio is quite confusing. We didn't document the design choices when writing this code, but the text-based output contains a table that was probably assumed to be the same as the DataGrid data. However, if you compare the DataGrid with this table, the table appears to be a flipped version of the DataGrid. This is probably why the rows were reversed.
DataGrids seem to follow the normal matrix conventions (but I could not find this anywhere in the documentation), with the smallest indices in the top left corner. The OpticStudio analysis output seems to follow the normal plot conventions, with the smallest indices at the bottom left corner. The text output can then be interpreted as a tabular representation of the plot. However, this is not true for all settings. If you show the PSF as True color, the Y-axis is not flipped.

To make things even more confusing: there is a "Center point is:" field in the text output, which gives an integer index of the center. Intuitively, I would interpret this as the one-based index of the center, relative to the top left corner. However, this value turns out to be located one row below the maximum value (which should be located at the center, if the PSF is symmetric). When assuming zero-based indexing, we arrive two rows below the maximum value. When interpreted as the one-based index from the bottom left, the "center point" is located at the maximum value.

We are working on a fix, but please note that the interpretation of the PSF output appears to be tricky. There is no reason to take the values returned by `zospy.analyses.psf.huygens_psf` for granted after we release a fix for this issue.

**Comment by andibarg** [comment]
Thank you for looking into it @crnh. 
I understand the fix is tricky because the OpticStudio output is confusing.

**Comment by crnh** [comment]
The fix in #56 has been merged and the new version is now available on PyPI!

**Comment by andibarg** [comment]
Thanks for the fix everyone!
